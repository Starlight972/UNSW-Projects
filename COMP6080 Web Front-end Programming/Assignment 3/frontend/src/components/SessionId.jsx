import React from 'react';
import { CopyToClipboard } from 'react-copy-to-clipboard';
import styled from 'styled-components';
import PropTypes from 'prop-types';

// MUI components
import Button from '@mui/material/Button';
import ContentCopyIcon from '@mui/icons-material/ContentCopy';

const SessionId = ({ sessionId }) => {
  return (
    <Wrapper>
      <SessionBtn
        variant="contained"
        color='secondary'
        id="session"
        sx={{ fontWeight: 800 }}
      >
        {`Session ID: ${sessionId}`}
      </SessionBtn>
      <CopyToClipboard text={`http://localhost:3000/player/join/${sessionId}`} id="copy-to-clipboard">
        <Button id='copy' variant="contained" >
          <ContentCopyIcon fontSize='small' />
        </Button>
      </CopyToClipboard>
    </Wrapper>
  )
}

export default SessionId;

SessionId.propTypes = {
  sessionId: PropTypes.any
}

const Wrapper = styled.div`
  display: flex;
  gap: 0.3rem;
  width: 200px;
  height: 50px;
`;

const SessionBtn = styled(Button)`
  width: 160px;
`;
