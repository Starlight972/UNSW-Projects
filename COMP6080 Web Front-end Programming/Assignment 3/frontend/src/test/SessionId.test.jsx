import React from 'react';
import { shallow } from 'enzyme';
import { CopyToClipboard } from 'react-copy-to-clipboard';

// MUI components
import Button from '@mui/material/Button';

// own component
import SessionId from '../components/SessionId';

describe('<SessionId />', () => {
  let wrapper;
  const sessionId = 1234;

  it('renders without crashing', () => {
    wrapper = shallow(<SessionId sessionId={sessionId} />);
  })

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have text', () => {
    expect(wrapper.find('#session').text()).toBe('Session ID: 1234');
  })

  it('should render one component', () => {
    expect(wrapper.find(Button)).toHaveLength(1);
    expect(wrapper.find(CopyToClipboard)).toHaveLength(1);
  })
})
