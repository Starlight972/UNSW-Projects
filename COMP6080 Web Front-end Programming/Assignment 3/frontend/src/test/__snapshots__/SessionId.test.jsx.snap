// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SessionId /> matches snapshot 1`] = `ShallowWrapper {}`;
